### DISCUSSION TITLE
accessing a specific chat ID

### DISCUSSION THREAD

**Discussion by realet68** [llm-workflow-engine/llm-workflow-engine#245] [Q&A]
with my apologies for the newbie question: 

in the Python API model, using: 

from chatgpt_wrapper import ChatGPT
...
bot = ChatGPT()
...
response = bot.ask(prompt)

Is there (a) an available method that will return the bot object's chat ID along with the chat completion message text, and (b) can I set the chat ID as a target of the "ask" method, in order to be able to take advantage of the context created within a specific chat across multiple sessions? 

Thanks! -Tom

**Comment by thehunmonkgroup** [comment]
I encourage you to investigate the relevant `GPTShell` code, it contains all the clues you need to perform most operations the shell can, but by leveraging the backend module directly.
